python/cudf/cudf/_lib/orc.pyx

```diff
@@ -62,9 +76,137 @@ from pyarrow.lib import NativeFile
 from cudf._lib.utils import _index_level_name, generate_pandas_metadata
 
 
-cpdef read_raw_orc_statistics(filepath_or_buffer):
+cdef _parse_column_type_statistics(column_statistics stats):
+    # Initialize stats to return and parse stats blob
+    column_stats = {}
+
+    if stats.number_of_values.has_value():
+        column_stats["number_of_values"] = stats.number_of_values.value()
+
+    if stats.has_null.has_value():
+        column_stats["has_null"] = stats.has_null.value()
+
+    cdef statistics_type type_specific_stats = stats.type_specific_stats
+
+    cdef integer_statistics* int_stats
+    cdef double_statistics* dbl_stats
+    cdef string_statistics* str_stats
+    cdef bucket_statistics* bucket_stats
+    cdef decimal_statistics* dec_stats
+    cdef date_statistics* date_stats
+    cdef binary_statistics* bin_stats
+    cdef timestamp_statistics* ts_stats
+
+    if holds_alternative[no_statistics](type_specific_stats):
+        return column_stats
+    elif int_stats := get_if[integer_statistics](&type_specific_stats):
+        if dereference(int_stats).minimum.has_value():
```

Comment:
All the `dereference` calls in this function are superfluous. Cython automatically knows to dereference pointers when you access. This is necessary for it to avoid introducing a separate syntax for  `->` that would make Cython code look even less Pythonic :)

Reply:
Sheesh. I always forget this. Cython is so verbose for C++ language features. Fixed now.